x/staking/keeper/unbonding_test.go

```diff
@@ -157,62 +157,80 @@ func doValidatorUnbonding(
-	require.True(t, found)
-	// Check that status is unbonded
-	require.Equal(t, types.BondStatus(1), validator.Status)
-}
-
-func TestValidatorUnbondingOnHold2(t *testing.T) {
-	var hookCalled bool
-	var ubdeID uint64
-	app, ctx, _, addrDels, addrVals := setup(t, &hookCalled, &ubdeID)
-	validator := doValidatorUnbonding(t, app, ctx, addrDels, addrVals, &hookCalled)
+	// Check that both validators have same unbonding time
+	require.Equal(t, validator1.UnbondingTime, validator2.UnbondingTime)
 
-	completionTime := validator.UnbondingTime
-	completionHeight := validator.UnbondingHeight
+	completionTime := validator1.UnbondingTime
+	completionHeight := validator1.UnbondingHeight
 
 	// PROVIDER CHAIN'S UNBONDING PERIOD ENDS - BUT UNBONDING CANNOT COMPLETE
 	ctx = ctx.WithBlockTime(completionTime.Add(time.Duration(1)))
 	ctx = ctx.WithBlockHeight(completionHeight + 1)
 	app.StakingKeeper.UnbondAllMatureValidators(ctx)
 
-	// Check that unbonding is not complete
-	validator, found := app.StakingKeeper.GetValidator(ctx, addrVals[0])
+	// Check that unbonding is not complete for both validators
+	validator1, found := app.StakingKeeper.GetValidator(ctx, addrVals[0])
 	require.True(t, found)
-	// Check that status is unbonding
-	require.Equal(t, types.BondStatus(2), validator.Status)
+	require.Equal(t, types.Unbonding, validator1.Status)
+	validator2, found = app.StakingKeeper.GetValidator(ctx, addrVals[1])
+	require.True(t, found)
+	require.Equal(t, types.Unbonding, validator2.Status)
+	unbondingVals := app.StakingKeeper.GetUnbondingValidators(ctx, completionTime, completionHeight)
+	require.Equal(t, 2, len(unbondingVals))
+	require.Equal(t, validator1.OperatorAddress, unbondingVals[0])
+	require.Equal(t, validator2.OperatorAddress, unbondingVals[1])
 
 	// CONSUMER CHAIN'S UNBONDING PERIOD ENDS - STOPPED UNBONDING CAN NOW COMPLETE
-	err := app.StakingKeeper.UnbondingCanComplete(ctx, ubdeID)
+	err := app.StakingKeeper.UnbondingCanComplete(ctx, ubdeIDs[0])
 	require.NoError(t, err)
 
-	validator, found = app.StakingKeeper.GetValidator(ctx, addrVals[0])
+	// Try again to unbond validators
+	app.StakingKeeper.UnbondAllMatureValidators(ctx)
+
+	// Check that unbonding is complete for validator1, but not for validator2
+	validator1, found = app.StakingKeeper.GetValidator(ctx, addrVals[0])
+	require.True(t, found)
+	require.Equal(t, types.Unbonded, validator1.Status)
+	validator2, found = app.StakingKeeper.GetValidator(ctx, addrVals[1])
+	require.True(t, found)
+	require.Equal(t, types.Unbonding, validator2.Status)
+	unbondingVals = app.StakingKeeper.GetUnbondingValidators(ctx, completionTime, completionHeight)
+	require.Equal(t, 1, len(unbondingVals))
+	require.Equal(t, validator2.OperatorAddress, unbondingVals[0])
+
+	// Unbonding for validator2 can complete
+	err = app.StakingKeeper.UnbondingCanComplete(ctx, ubdeIDs[1])
+	require.NoError(t, err)
+
+	// Try again to unbond validators
+	app.StakingKeeper.UnbondAllMatureValidators(ctx)
```

Comment:
What triggers this to be called IRL?

Reply:
The next block. At any given time, `app.StakingKeeper.UnbondAllMatureValidators` will be eventually called again as long as the chain is producing new blocks (i.e., liveness is not violated).